server/service/src/sensor/berlinger.rs

```diff
@@ -0,0 +1,380 @@
+use chrono::NaiveDateTime;
+use repository::{DatetimeFilter, EqualFilter};
+use repository::{
+    RepositoryError, Sensor, SensorFilter, SensorRepository, SensorRow, SensorRowRepository,
+    SensorType, StorageConnection, TemperatureBreach, TemperatureBreachConfig,
+    TemperatureBreachConfigFilter, TemperatureBreachConfigRepository, TemperatureBreachConfigRow,
+    TemperatureBreachConfigRowRepository, TemperatureBreachFilter, TemperatureBreachRepository,
+    TemperatureBreachRow, TemperatureBreachRowRepository, TemperatureBreachRowType, TemperatureLog,
+    TemperatureLogFilter, TemperatureLogRepository, TemperatureLogRow, TemperatureLogRowRepository,
+};
+use util::uuid::uuid;
+
+use crate::{service_provider::ServiceContext, SingleRecordError};
+
+extern crate temperature_sensor;
+use temperature_sensor::*;
+
+pub fn get_breach_row_type(breach_type: &BreachType) -> TemperatureBreachRowType {
+    match breach_type {
+        BreachType::ColdConsecutive => TemperatureBreachRowType::ColdConsecutive,
+        BreachType::ColdCumulative => TemperatureBreachRowType::ColdCumulative,
+        BreachType::HotConsecutive => TemperatureBreachRowType::HotConsecutive,
+        BreachType::HotCumulative => TemperatureBreachRowType::HotCumulative,
+    }
+}
+
+pub fn get_matching_sensor_serial(
+    connection: &StorageConnection,
+    serial: &str,
+) -> Result<Vec<Sensor>, RepositoryError> {
+    SensorRepository::new(connection)
+        .query_by_filter(SensorFilter::new().serial(EqualFilter::equal_to(&serial)))
+}
+
+pub fn get_matching_sensor_log(
```

Comment:
Thanks @adrianwb, i really like these methods and their names, I think it makes for great abstraction, and top level method (the one that consumes them) reads very well